src/lint/lint.ts

```diff
@@ -48,37 +51,51 @@ export async function lint(options: Options, config: ProjectConfig) {
     minimumSeverity: Severity.WARNING
   });
 
-
   const analyzer = new Analyzer({
     urlLoader: new FSUrlLoader(config.root),
     urlResolver: new PackageUrlResolver(),
   });
-
   const linter = new lintLib.Linter(rules, analyzer);
 
+  // This is awkward, it would be nice to be able to get access to the analysis
+  // of a lint run rather than doing this sort of parallel construction.
+  // We want the analysis that the linter sees so that we can apply fixes
+  // to the same version of the files that the fixes were constructed for.
   let warnings;
+  let analysis;
   if (options.input) {
-    warnings = await linter.lint(options.input);
+    [warnings, analysis] = await Promise.all(
+        [linter.lint(options.input), analyzer.analyze(options.input)]);
   } else {
-    warnings = await linter.lintPackage();
+    [warnings, analysis] =
+        await Promise.all([linter.lintPackage(), analyzer.analyzePackage()]);
   }
 
   const filtered = warnings.filter((w) => !filter.shouldIgnore(w));
 
-  const printer = new WarningPrinter(
-      process.stdout, {verbosity: 'full', color: true});
-  await printer.printWarnings(filtered);
+  if (options.fix) {
+    return fix(filtered, options, config, analyzer, analysis);
+  } else {
+    return report(filtered);
+  }
+}
+
+async function report(warnings: Warning[]) {
+  const printer =
+      new WarningPrinter(process.stdout, {verbosity: 'full', color: true});
+  await printer.printWarnings(warnings);
 
-  if (filtered.length > 0) {
+  if (warnings.length > 0) {
     let message = '';
-    const errors = filtered.filter((w) => w.severity === Severity.ERROR);
-    const warnings = filtered.filter((w) => w.severity === Severity.WARNING);
-    const infos = filtered.filter((w) => w.severity === Severity.INFO);
+    const errors = warnings.filter((w) => w.severity === Severity.ERROR);
+    const warningLevel =
```

Comment:
why the name change? What's a warning level?

Reply:
It shadowed the other `warnings` variable. Clarified.